Get SchwiftyTests/CompleterTests.swift

```diff
@@ -0,0 +1,2477 @@
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "atest"
+        let expectedRange = NSMakeRange(1, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_AtTheEnd_InsertedNormally() {
+        let text = "test"
+        let range = NSMakeRange(4, 0) // End
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "testa"
+        let expectedRange = NSMakeRange(5, 0) // End
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_InTheMiddleOfWhitespace_InsertedNormally() {
+        let text = "tes\t  \tt"
+        let range = NSMakeRange(5, 0) // After the first space
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes\t a \tt"
+        let expectedRange = NSMakeRange(6, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_BetweenTabs_InsertedNormally() {
+        let text = "tes\t\tt"
+        let range = NSMakeRange(4, 0) // Between the tabs
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes\ta\tt"
+        let expectedRange = NSMakeRange(5, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_BetweenSpaces_InsertedNormally() {
+        let text = "tes  t"
+        let range = NSMakeRange(4, 0) // After the first space
+        
+        let insertion = "a"
+        let (newText, newRange) = Completer.completedText(for: insertion, in: text, range: range)
+        
+        let expectedText = "tes a t"
+        let expectedRange = NSMakeRange(5, 0) // After the "a"
+        
+        XCTAssertEqual(newText, expectedText)
+        XCTAssertEqual(newRange.location, expectedRange.location)
+        XCTAssertEqual(newRange.length, expectedRange.length)
+    }
+    
+    func test_NormalCharacter_WithEmojis_InsertedNormally() {
+        let text = "👱🏾🚀test"
```

Comment:
🤡 This is a good example why we need utf16. With characters, the cursor position would be off after entering this text.